tiledb/sm/filesystem/win.cc

```diff
@@ -74,13 +76,39 @@ std::string get_last_error_msg() {
           (LPTSTR)&lpMsgBuf,
           0,
           NULL) == 0) {
-    LocalFree(lpMsgBuf);
+    if (lpMsgBuf) {
+      LocalFree(lpMsgBuf);
+    }
     return "unknown error";
   }
   std::string msg(reinterpret_cast<char*>(lpMsgBuf));
   LocalFree(lpMsgBuf);
   return msg;
 }
+
+std::string get_last_error_msg(
+    decltype(GetLastError()) gle, const TCHAR* func_desc = "") {
+  std::string gle_desc = get_last_error_msg_desc(gle);
+
+  auto buf_len = gle_desc.length() + std::strlen(func_desc) + 50;
+  auto deleter = [](char* p) { tdb_delete_array(p); };
+  auto display_buf = shared_ptr<char>(tdb_new_array(char, buf_len), deleter);
```

Comment:
I tried, std::make_unique gets flagged by our heap violation checks, unique_ptr was not functional in our dynamic ecosystem for an array (at least when I last tried here), hence fell back to shared_ptr which doesn't leave the routine, rather than add another exception to our find_heap_api_violations.py script.

Reply:
Please add the exception to the checker.

Reply:
Changed back to use std::unique_ptr, added exception to checker.